### PR TITLE
[IMP] show discount if set on pricelist

### PR DESCRIPTION
**Pull request by bguillot**

### CODE REVIEW

File: shopinvader/models/shopinvader_variant.py [hunk 1]
```diff
@@ -142,10 +142,19 @@ def _get_price_per_qty(self, qty, pricelist, fposition, company=None):
         tax_included = tax_id.price_include
         value = self.env['account.tax']._fix_tax_included_price_company(
             final_price, product.taxes_id, tax_id, company)
-        return {
+        res = {
             'value': value,
             'tax_included': tax_included
         }
+        if pricelist.discount_policy == 'without_discount':
```
Comment (lmignon) [inline shopinvader/models/shopinvader_variant.py:149]: @bguillot IMO this change requires a unittest....

**Review by lmignon — CHANGES_REQUESTED** [review]

File: shopinvader/models/shopinvader_variant.py [hunk 2]
```diff
@@ -142,10 +142,19 @@ def _get_price_per_qty(self, qty, pricelist, fposition, company=None):
...
+            new_list_price, currency_id = self.env['sale.order.line']._get_real_price_currency(product, rule_id, qty or 1.0, product.uom_id, pricelist.id)
```
Comment (lmignon) [inline shopinvader/models/shopinvader_variant.py:150]: Is the tax included or not into this new_list_price? 
We must also take into account the fiscal_position. 
I've the feeling that the result of this method will be wrong... But I hope I'm wrong...

**Review by lmignon — CHANGES_REQUESTED** [review]

File: shopinvader/models/shopinvader_variant.py (hunk 2 above)
Comment (lmignon) [reply shopinvader/models/shopinvader_variant.py:150]: I'm now sure I'm right :smirk: see https://github.com/akretion/odoo-shopinvader/pull/207/commits/59054bde505da541d53c776942e5c84813563f23

**Comment by lmignon** [comment]
@sebastienbeau @bguillot This once can be closed. I've fixed the code to take into account the fiscal postion (tax mapping) and the digital precision to apply to discount in the computation of the original price and discount. I've also added tests to cover the different cases when computing the price (price list with and without discount, price list with and wihtout discount policy, fiscal position with and without tax mapping) see #207 and https://github.com/akretion/odoo-shopinvader/pull/207/commits/59054bde505da541d53c776942e5c84813563f23

**Comment by sebastienbeau** [comment]
Close pull request as this have been done here : https://github.com/akretion/odoo-shopinvader/pull/207

**Pull request closed without merging**